packages/editor/src/components/block-list/style.scss

```diff
@@ -820,39 +845,50 @@
 	}
 
 	.editor-block-list__block[data-align="left"] & {
-		@include break-small() {
-			// RTL note: this rule should not be auto-flipped based on direction.
-			/*rtl:ignore*/
-			float: left;
-		}
+		// RTL note: this rule should not be auto-flipped based on direction.
+		/*rtl:ignore*/
+		float: left;
 	}
 
+	.editor-block-list__block[data-align="left"] &,
 	.editor-block-list__block[data-align="right"] & {
-		@include break-small() {
-			// RTL note: this rule should not be auto-flipped based on direction.
-			/*rtl:ignore*/
-			float: right;
-		}
+		// Move the block toolbar out of the flow using translate, but less for floats.
+		transform: translateY(-$block-padding -$border-width);
+	}
+
+	.editor-block-list__block[data-align="right"] & {
+		// RTL note: this rule should not be auto-flipped based on direction.
+		/*rtl:ignore*/
+		float: right;
 	}
 }
 
 // Position the block toolbar when contextual.
 .editor-block-contextual-toolbar .editor-block-toolbar {
 	width: 100%;
 
-	// Hide right border on desktop, where the .components-toolbar instead has a right border.
 	@include break-small() {
+		width: auto;
+
+		// Hide right border on desktop, where the .components-toolbar instead has a right border.
 		border-right: none;
-	}
 
-	// This prevents floats from messing up the position.
-	@include break-small() {
+		// This prevents floats from messing up the position.
 		position: absolute;
 		left: 0;
 	}
+}
 
-	@include break-small() {
-		width: auto;
+// This rule ensures that any blocks that are not a Paragraph, that follows any aligned block, clears them.
```

Comment:
I don't like that we are creating special rules on a per-block basis. 

In this case, it was the only way I could think of, to ensure that the following worked as intended:

<img width="740" alt="screenshot 2018-11-02 at 09 45 57" src="https://user-images.githubusercontent.com/1204802/47903551-299cee80-de84-11e8-9898-3403d80b7093.png">

To elaborate — unless this bottom placeholder block explicitly _clears_ the preceding floats, then the block outline will be painted all the way up to before the first float. This is because the block techically floats after the first two, so the outline starts all the way at the top. Except this is not the user expectation.

On the flipside, we explicitly allow the Paragraph block to not clear the preceding floats, so that this can work:

<img width="678" alt="screenshot 2018-11-02 at 09 47 58" src="https://user-images.githubusercontent.com/1204802/47903783-6f59b700-de84-11e8-8e52-370c78db0ab8.png">

This is the normal floats behavior, and it "looks correct" for the user because the text snakes in between the floats.